File: src/main/java/com/hris/HRIS/model/EmployeePayItemModel.java
package com.hris.HRIS.model;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Getter
@Setter
@ToString

@Document(collection = "employee-payitem")
public class EmployeePayItemModel {
    @Id
    private String id;
    private String payItemId;
    private String email;
    private double amount;
}

File: client/src/app/shared/data-models/employee-payitem.model.ts
import { Injectable } from "@angular/core";
import { PayItemModel } from "./payitem.model";

@Injectable({
    providedIn: 'root'
})
export class EmployeePayItemModel{
    id: String = "";
    payItemId: String = "";
    payitem: PayItemModel = new PayItemModel();
    email: String = "";
    type: String = "";
    value: number = 0.0;
    amount: number = 0.0;
}